vars/withCustomCommitStates.groovy

```diff
@@ -0,0 +1,55 @@
+import com.cloudbees.groovy.cps.NonCPS
+
+def call(scm, final String credentialsId, final String context, block) {
+
+  def STATE_NAME_PENDING = 'pending'
+  def STATE_NAME_SUCCESS = 'success'
+  def STATE_NAME_FAILURE = 'failure'
+  def STATE_NAME_ERROR = 'error'
+
+  def STATES = [
+    [state: STATE_NAME_PENDING, description: 'The build is pending'],
+    [state: STATE_NAME_SUCCESS, description: 'This commit looks good'],
+    [state: STATE_NAME_FAILURE, description: 'An error or test failure occurred'],
+    [state: STATE_NAME_ERROR, description: 'This commit cannot be built']
+  ]
+
+  def PENDING_STATE = STATES.find{it['state'] == STATE_NAME_PENDING}
+  def SUCCESS_STATE = STATES.find{it['state'] == STATE_NAME_SUCCESS}
+  def FAILURE_STATE = STATES.find{it['state'] == STATE_NAME_FAILURE}
+
+  def currentState = FAILURE_STATE
+  try {
+    setCommitState(scm, credentialsId, PENDING_STATE['state'], PENDING_STATE['description'], context)
+    block()
```

Comment:
Just for a second I thought this means "Block this method for some time", it's tiny thing, maybe renaming to something like `pipelineBody()` or anything similar ?

Reply:
good point, I've been using this because of a tutorial, were they call Closures blocks

Reply:
haha, it too me also 2min to figure out `block()`  - i thought that the code blocks for some reason. 

The better name is `codeBlock()`